common/models/user.js

```diff
@@ -434,6 +441,126 @@ module.exports = function(User) {
+   *   });
+   * }
+   * ```
+   *
+   * Usually you should only require to modify a subset of these params
+   * See `User.verify()` and `User.prototype.verify()` doc for params reference
+   */
+
+  User.getVerifyOptions = function() {
+    return this.settings.verifyOptions || {type: 'email'};
+  };
+
+  /**
+   * Verify a user's identity by sending them a confirmation email.
+   *
+   * ```js
+   *    var verifyOptions = {
+   *      type: 'email',
+   *      to: user.email,
+   *      template: 'verify.ejs',
+   *      templateFn: function(options, cb) { cb(...) }
+   *      redirect: '/',
+   *      tokenGenerator: function (user, cb) { cb(...); }
+   *    };
+   *
+   *    user.verify(options, next);
+   * ```
+   *
+   * @param {*} id The user id
+   * @options {Object} verifyOptions
+   * @property {String} type Must be 'email'.
+   * @property {String} to Email address to which verification email is sent.
+   * @property {String} from Sender email addresss, for example
+   *   `'noreply@myapp.com'`.
+   * @property {String} subject Subject line text.
+   * @property {String} text Text of email.
+   * @property {String} template Name of template that displays verification
+   *  page, for example, `'verify.ejs'.
+   * @property {Function} templateFn A function generating the email HTML body
+   * from `verify()` options object and generated attributes like `options.verifyHref`.
+   * It must accept the option object and a callback function with `(err, html)`
+   * as parameters
+   * @property {String} redirect Page to which user will be redirected after
+   *  they verify their email, for example `'/'` for root URI.
+   * @property {Function} generateVerificationToken A function to be used to
+   *  generate the verification token. It must accept the user object and a
+   *  callback function. This function should NOT add the token to the user
+   *  object, instead simply execute the callback with the token! User saving
+   *  and email sending will be handled in the `verify()` method.
+   * @callback {Function} fn Callback function.
+   * @param {Error} err Error object.
+   * @param {Object} object Contains email, token, uid.
+   * @promise
+   */
+
+  User.verify = function(uid, verifyOptions, options, cb) {
+    if (cb === undefined && typeof options === 'function') {
+      cb = options;
+      options = undefined;
+    }
+    cb = cb || utils.createPromiseCallback();
+    if (options === undefined) {
+      options = verifyOptions;
+      verifyOptions = undefined;
+    }
+    verifyOptions = verifyOptions || this.getVerifyOptions();
+
+    // Make sure to use the constructor of the (sub)class
+    // where the method is invoked from (`this` instead of `User`)
+    this.findById(uid, options, (err, user) => {
+      if (err) return cb(err);
+
+      if (!user) {
+        const err = new Error(`User ${uid} not found`);
+        Object.assign(err, {
+          code: 'USER_NOT_FOUND',
+          statusCode: 422, // or 404?
```

Comment:
It depends on the URL. I would personally expose this new method as `http: {verb: 'post', path: '/:uid/verify'}`, in which case 404 makes perfect sense.